compiler/pipes/optimization.cpp

```diff
@@ -316,6 +316,19 @@ VertexPtr OptimizationPass::on_enter_vertex(VertexPtr root) {
     }
   } else if (auto op_conv_string_vertex = root.try_as<op_conv_string>()) {
     root = convert_strval_to_magic_tostring_method_call(op_conv_string_vertex);
+  } else if (root->type() == op_function) {
+    if (current_function->type == FunctionData::func_class_holder) {
+      current_function->class_id->members.for_each([this](ClassMemberInstanceField &class_field) {
+        if (class_field.var->init_val) {
+          if (auto var_vertex = class_field.var->init_val.try_as<op_var>()) {
+            if (vk::any_of_equal(var_vertex->var_id->init_val->type(), op_string_build, op_concat)) {
+              auto &op_concat_vertex = var_vertex->var_id->init_val;
+              op_concat_vertex = optimize_string_building(op_concat_vertex);
+            }
+          }
+        }
+      });
+    }
```

Comment:
This is 100% a wrong place.
I suppose, that the only change you need to do for fixing the issue, is adding
```
        run_function_pass(class_field.var->init_val, this);
```
into `void OptimizationPass::on_finish()` at the top of `if (class_field.var->init_val)`.
This test passes after this change.
Please, check that vkcom still compiles and that vkcom codegen diff is expected.

Reply:
fixed